### PR TITLE
feat(ee): Add copilot integration (v1) to the conversation sidebar

### PR DESCRIPTION
**Pull request by pranavrajs**
Screenshot 2024-12-10 at 2 16 40 AM

### CODE REVIEW

File: app/javascript/dashboard/components-next/copilot/Copilot.story.vue
Comment (iamsivin) [inline app/javascript/dashboard/components-next/copilot/Copilot.story.vue:1]: @pranavrajs I guess this story is not working as expected. Can you verify it?

https://github.com/user-attachments/assets/f4ad969b-3d12-43c7-a3a3-caa1c5d9711a

**Review by iamsivin — APPROVED** [review]
LGTM. I added some minor comments. Feel free to ignore them if they’re not needed.

**Review by sojan-official — APPROVED** [review]
Backend changes LGTM

File: app/javascript/dashboard/components-next/copilot/Copilot.vue [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  >
+    <div ref="chatContainer" class="flex-1 overflow-y-auto py-4 space-y-6 px-4">
+      <template v-for="message in messages" :key="message.id">
+        <CopilotAgentMessage
+          v-if="message.role === 'user'"
+          :support-agent="supportAgent"
+          :message="message"
+        />
+        <CopilotAssistantMessage
+          v-else-if="['assistant', 'system'].includes(message.role)"
```
[the remark this answers is not in the document]
Comment (pranavrajs) [reply app/javascript/dashboard/components-next/copilot/Copilot.vue]: Done

File: app/javascript/dashboard/components-next/copilot/Copilot.story.vue
Comment (pranavrajs) [reply app/javascript/dashboard/components-next/copilot/Copilot.story.vue:1]: Fixed, good catch.

File: app/javascript/dashboard/components-next/copilot/CopilotAgentMessage.vue [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    required: true,
+  },
+});
+</script>
+
+<template>
+  <div class="flex flex-row gap-2">
+    <Avatar
+      :name="supportAgent.available_name"
+      :src="supportAgent.avatar_url"
```
[the remark this answers is not in the document]
Comment (pranavrajs) [reply app/javascript/dashboard/components-next/copilot/CopilotAgentMessage.vue:20]: I tried to update it, but there was an issue with the computed properties. I couldn't get why it was breaking. I’ll fix this in a followup PR.

**Pull request merged**